src/common/src/types/mod.rs

```diff
@@ -302,6 +302,25 @@ impl DataType {
         )
     }
 
+    pub fn is_scalar(&self) -> bool {
+        matches!(
+            self,
+            DataType::Boolean
+                | DataType::Int16
+                | DataType::Int32
+                | DataType::Int64
+                | DataType::Decimal
+                | DataType::Float32
+                | DataType::Float64
+                | DataType::Varchar
+                | DataType::Date
+                | DataType::Timestamp
+                | DataType::Timestampz
+                | DataType::Time
+                | DataType::Interval
+        )
```

Comment:
Okay. Thank you for pointing this out. I will keep this function for now and use it as a wrapper for your suggestion to make access a bit easier.

Reply:
It is okay to keep the wrapper, but `DataType` should call `DataTypeName`, rather than the reverse. It is always doable to `DataTypeName::from(DataType)`, but sometimes `DataType::from(DataTypeName)` panics.